piker/data/_ahab.py

```diff
@@ -124,7 +124,9 @@ def __init__(
 
     async def process_logs_until(
         self,
-        patt: str,
+        # this is a predicate func for matching log msgs emitted by the
+        # underlying containerized app
+        patt_matcher: Callable[[str], bool],
```

Comment:
I think the typing here is techincally an `Awaitable` now, but it's not a big deal and i don't want this to delay the patch.